crates/matrix-sdk-crypto/src/store/memorystore.rs

```diff
@@ -103,36 +103,9 @@ pub struct MemoryStore {
     dehydrated_device_pickle_key: RwLock<Option<DehydratedDeviceKey>>,
     next_batch_token: RwLock<Option<String>>,
     room_settings: StdRwLock<HashMap<OwnedRoomId, RoomSettings>>,
-    save_changes_lock: Arc<Mutex<()>>,
-}
+    room_key_bundles: StdRwLock<HashMap<OwnedRoomId, Vec<StoredRoomKeyBundleData>>>,
```

Comment:
Probably not important enough to care about this, but I find it a bit surprising that we're not keying this by the user ID as well.

Then, when loading the bundle, we have to search for it, if the same user adds multiple bundles, we have bundles that we'll never fetch from the store.